rust/src/fireblocks/mod.rs

```diff
@@ -0,0 +1,791 @@
+//! Fireblocks API signer integration
+
+mod jwt;
+mod types;
+
+use crate::sdk_adapter::{Pubkey, Signature, Transaction};
+pub use crate::traits::SignedTransaction;
+use crate::{error::SignerError, traits::SolanaSigner, transaction_util::TransactionUtil};
+use base64::{engine::general_purpose::STANDARD, Engine};
+use std::str::FromStr;
+use types::{
+    CreateTransactionRequest, CreateTransactionResponse, ExtraParameters,
+    ProgramCallExtraParameters, RawExtraParameters, RawMessage, RawMessageData,
+    TransactionResponse, TransactionSource, VaultAddressesResponse,
+};
+
+/// Fireblocks-based signer using Fireblocks' API
+#[derive(Clone)]
+pub struct FireblocksSigner {
+    api_key: String,
+    private_key_pem: String,
+    vault_account_id: String,
+    asset_id: String,
+    public_key: Pubkey,
+    api_base_url: String,
+    client: reqwest::Client,
+    poll_interval_ms: u64,
+    max_poll_attempts: u32,
+    use_program_call: bool,
+}
+
+impl std::fmt::Debug for FireblocksSigner {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("FireblocksSigner")
+            .field("public_key", &self.public_key)
+            .field("vault_account_id", &self.vault_account_id)
+            .field("asset_id", &self.asset_id)
+            .field("use_program_call", &self.use_program_call)
+            .finish_non_exhaustive()
+    }
+}
+
+/// Configuration for creating a FireblocksSigner
+#[derive(Clone)]
+pub struct FireblocksSignerConfig {
+    pub api_key: String,
+    pub private_key_pem: String,
+    pub vault_account_id: String,
+    /// Asset ID (default: "SOL", use "SOL_TEST" for devnet)
+    pub asset_id: Option<String>,
+    pub api_base_url: Option<String>,
+    pub poll_interval_ms: Option<u64>,
+    pub max_poll_attempts: Option<u32>,
+    /// Use PROGRAM_CALL operation for transaction signing (auto-broadcasts to Solana).
+    /// Default: false (uses RAW signing)
+    pub use_program_call: Option<bool>,
+}
+
+impl FireblocksSigner {
+    /// Create a new FireblocksSigner
+    ///
+    /// Note: You must call `init()` after construction to fetch the public key.
+    ///
+    /// # Arguments
+    ///
+    /// * `config` - Configuration for the signer
+    pub fn new(config: FireblocksSignerConfig) -> Self {
+        Self {
+            api_key: config.api_key,
+            private_key_pem: config.private_key_pem,
+            vault_account_id: config.vault_account_id,
+            asset_id: config.asset_id.unwrap_or_else(|| "SOL".to_string()),
+            public_key: Pubkey::default(),
+            api_base_url: config
+                .api_base_url
+                .unwrap_or_else(|| "https://api.fireblocks.io".to_string()),
+            client: reqwest::Client::new(),
+            poll_interval_ms: config.poll_interval_ms.unwrap_or(1000),
+            max_poll_attempts: config.max_poll_attempts.unwrap_or(60),
```

Comment:
default should maybe be longer.. maybe we default to 5m? it's not unusual for this to take a decent amount of time.